src/sentry/demo/demo_start.py

```diff
@@ -78,7 +78,23 @@ def post(self, request):
             logger.info("post.assigned_org", extra={"organization_slug": org.slug})
 
         auth.login(request, user)
-        resp = self.redirect(get_redirect_url(request, org))
+
+        extra_query_string = request.POST.get("extraQueryString")
+        redirect_url = get_redirect_url(request, org)
+
+        hash_param = ""
+
+        if "#" in redirect_url:
+            partition = redirect_url.rpartition("#")
+            redirect_url = partition[0]
+            hash_param = partition[1] + partition[2]
+
+        if "/?" in redirect_url:
```

Comment:
@roggenkemper why are we checking for `/?` instead of `?`? I'm slightly concerned someone might forget a `/` at the end of the URL and this wouldn't work.

Reply:
makes sense. ill change it to just ?

Reply:
@roggenkemper I think so. I don't think there would ever be cases where we have an escaped `?` in the URL so it should be fine.